### PR TITLE
(fix):Object.normaliseMail: Cannot read properties of null

### PR DESCRIPTION
**Pull request by dobri1408**
I found out that if somehow it happens that a link has a null value, it can be caused by a faulty copy-paste, this error can happen. This is due to the fact that the "email"/"telephone" is null or "res.url" is null.
Captură de ecran din 2023-03-16 la 14 54 56

### CODE REVIEW

**Review by tiberiuichim — APPROVED** [review]
Would it be possible to add this test case in the jest test?

**Comment by dobri1408** [comment]
Hello @tiberiuichim, sure, thanks!

**Comment by dobri1408** [comment]
Hi, I added the tests. Please, let me know if anything else should be changed or added.

File: src/helpers/Url/Url.js [hunk 1]
```diff
@@ -300,6 +300,7 @@ export function checkAndNormalizeUrl(url) {
         res.isValid = false;
       }
     }
+    if (res.url == undefined || res.url == null) res.isValid = false;
```
Comment (tiberiuichim) [inline src/helpers/Url/Url.js]:
```suggestion
    if (typeof res.url === "undefined" || res.url === null) res.isValid = false;
```

Comment (tiberiuichim) [reply src/helpers/Url/Url.js]: Although I see that with modern JS it's possible to compare something to `undefined`.

**Comment by tiberiuichim** [comment]
@dobri1408 please use triple equality operator

**Comment by tiberiuichim** [comment]
I've had to make this customization to make it work: https://github.com/eea/volto-cca-policy/blob/d92a80db622999357310ec9196ee30734789c3c0/src/customizations/volto/components/manage/AnchorPlugin/components/LinkButton/AddLinkForm.jsx

**Comment by dobri1408** [comment]
I fixed the error that caused the internal links to not work anymore.

**Review by sneridagh — APPROVED** [review]

**Pull request merged**